index.d.ts

```diff
@@ -56,7 +56,7 @@ export interface ApiParams {
 
 export type ApiCallback = (err?: Error|string|object|null, result?: any) => void;
 
-export type ApiFunction = (p: ApiParams, callback?: ApiCallback) => any;
+export type ApiFunction = (p: ApiParams, callback?: ApiCallback) => Promise<any>;
```

Comment:
@epheph Wanna try this?!

Reply:
We chatted in person. This doesn't work with what we are trying to do because we still return promise when a callback is present. I figure it can be like db.asCallback, you still return the promise for them to ignore?